attest/eventlog.go

```diff
@@ -0,0 +1,399 @@
+package attest
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"sort"
+
+	// Ensure hashes are available.
+	_ "crypto/sha256"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+// TPM algorithms. See the TPM 2.0 specification section 6.3.
+//
+// https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf#page=42
+const (
+	algSHA1   uint16 = 0x0004
```

Comment:
Should/could we use the constants from tpm2?

ie: https://github.com/google/go-attestation/blob/a801f7333b9e043c39ea938c44884b05fab9691f/verifier/quote.go#L18

Reply:
Changing this to use tpm2.Algorithm. Going to keep the switch statement instead of using tpm2.Algorithm.HashConstructor(). If someone is using something other than SHA1/SHA256 I want to know and purposefully add it.